### DISCUSSION TITLE
power lines not working

### DISCUSSION THREAD

**Discussion by AtMostafa** [Karpova-Lab/pyControl-D-Series-Breakout#4] [Q&A] [answered]
Hi,

I had problems with some of the digital outputs, so I tested each line (by toggling it every second) and realized that NONE of the power lines work (they only show ~0.2V difference in their on/off levels). I double-checked the part number for U3,4,5 and the solders, but they all seem to be fine.

I was just wondering if there are any known issues with the driver ICs or if anyone has encountered the same problem.

Bests,
Mostafa

**Comment by alustig3** [comment, marked as answer]
U3,4,5 are low side driver ICs that will sink current. To power a motor or solenoid you would supply the motor input with 12v for instance, and then instead of connecting the motor's ground to common ground, you connect it into the power driver IC. 

If you want to use those power lines as digital outputs you can connect a pullup resistor to 5V, or if you need 3V logic you could first create a voltage divider that gets the 5V down to 3V, and then add a pullup resistor between your new 3V and the power line.

A good explanation can be found in the pyControl documentation here: https://pycontrol.readthedocs.io/en/latest/user-guide/hardware/#behaviour-ports

**Reply by AtMostafa** [reply]
Ahhh... I just assumed the power lines are like relays, sourcing higher currents. Thanks ndy for clearing that up :)
